### ISSUE TITLE
Automatic testing for compiler backward compatibility

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#14206]
We're currently not testing that a new compiler works with older versions of the standard library. Such a test would've been able to detect and prevent #14194.

We should set up an automated test to ensure compiler changes are backwards compatible with older versions of the standard library.

In [`x86_64-gnu-test`](https://github.com/crystal-lang/crystal/blob/0b57b6cc8e4f5003267549892d23d2af5ad6309b/.github/workflows/linux.yml#L14) we're testing that the current stdlib and compiler *build* with an older compiler version. Now we also need the other way around.
It would probably be sufficient to test against the outer limits, i.e. the previous release and the lowest supported release (at this point: 1.0).

**Comment by HertzDevil** [comment]
For reference, the LLVM intrinsic name issue was mentioned in #13348

**Comment by Fryguy** [comment]
Would it help to regularly run https://github.com/crystal-lang/bootstrap-script ?

**Comment by HertzDevil** [comment]
That's only for forward builds, not backward

**Comment by straight-shoota** [comment]
Yeah, and we're already testing this in [`x86_64-gnu-test`](https://github.com/crystal-lang/crystal/blob/0b57b6cc8e4f5003267549892d23d2af5ad6309b/.github/workflows/linux.yml#L14).
